File: common/src/main/kotlin/cn/jawbreakers/candycraftce/block/SameBlockCullBlock.java
package cn.jawbreakers.candycraftce.block;

import net.minecraft.core.Direction;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.minecraft.world.level.block.state.BlockState;

public class SameBlockCullBlock extends Block {
	public SameBlockCullBlock(Properties properties) {
		super(properties);
	}

	@Override
	public boolean skipRendering(BlockState state, BlockState adjacentState, Direction side) {
		return adjacentState.is(this) || super.skipRendering(state, adjacentState, side);
	}
}

File: common/src/main/kotlin/cn/jawbreakers/candycraftce/block/SameBlockCullBlock.kt
package cn.jawbreakers.candycraftce.block

import net.minecraft.core.Direction
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.state.BlockState

open class SameBlockCullBlock(properties: Properties) : Block(properties) {
    @Suppress("DEPRECATION")
    @Deprecated("Deprecated in Java")
    override fun skipRendering(state: BlockState, adjacentState: BlockState, side: Direction): Boolean {
        return adjacentState.`is`(this) || super.skipRendering(state, adjacentState, side)
    }
}
